### ISSUE TITLE
Show a subset of mask pixels

### ISSUE THREAD

**Issue by kadrlica** [LSSTDESC/desc-exp-checker#25]
We would like to show only a subset of the masked pixels. For LSST, the mask bit values can be found with following the instructions [here](https://pipelines.lsst.io/v/DM-11392/getting-started/display.html#interpreting-displayed-mask-colors).
```
for maskName, maskBit in sorted(mask.getMaskPlaneDict().items(), key=lambda i: (i[1],i[0])):
    print('{}: {} ({})'.format(maskBit, maskName, display.getMaskPlaneColor(maskName)))    

0: BAD (red)
1: SAT (green)
2: INTRP (green)
3: CR (magenta)
4: EDGE (yellow)
5: DETECTED (blue)
6: DETECTED_NEGATIVE (cyan)
7: SUSPECT (yellow)
8: NO_DATA (orange)
```
I think that a minimal set of bits to start with are
```
0: BAD (red)
1: SAT (green)
3: CR (magenta)
4: EDGE (yellow)
8: NO_DATA (orange)
```
I think that we either want to do a multi-layer mask (with corresponding alpha values) or do a ranked ordering of bits (the second is easier but less flexible).

The current implementation of the mask coloring can be found [here](https://github.com/LSSTDESC/desc-exp-checker/blob/e9c75042caa246f467ff68d2ab6c15c6cb23a5db/assets/webfits-canvas.0.3.1.js#L201-L215).

**Comment by rmjarvis** [comment]
I think I would add 2: INTRP (these are pixels whose original data is not being used, and the data is being interpolated, although probably these are always also some other bit as well) and 7: SUSPECT (not completely sure about this one, since I don't know how DM decides what is suspect, but it sounds like what I would consider a "bad for weak lensing" pixel).  

So basically everything except the DETECTED pixels.  (I don't really understand DETECTED_NEGATIVE, so not sure whether that is better to include or not.)

Also, I'm happy enough with collapsing the colors into a single color for the image viewer representing any pixel that is suspicious or worse.

**Comment by kadrlica** [comment]
I'm going to push back on this. I think that we want to avoid displaying masked bits for pixels that look "good". We want eye ballers to be able to identify improper masking of good pixels. This is impossible if we are showing masks for a large sets of pixels that appear good.

For this reason, I would not include the INTRP pixels because if everything is working properly then the underlying pixels will appear good. In contrast, if the INTRP algorithm is messing up, we would like the eyeballers to catch it and not to think "Oh, it's already being masked, so we are ok". (I could go off on a tangent about how interpolation makes it really hard to check the performance of algorithms, but I'll save that fight for another day.)

Similarly, experience with DES has me hesitant about including SUSPECT bits in the mask. We use SUSPECT for a lot of stuff that is below the level that any eyeball checker will be able to see (for example, small extra noise in CCD 57). With DES, the SUSPECT bits often look a lot like a masking algorithm has gone off the rails.

Basically, I think that the mask plane we show should be flagging eyeballable (or otherwise standard such as EDGE) features. I'm pushing on this because we need keep the exposure checker optimized for identifying features in the data.

**Comment by kadrlica** [comment]
@rmjarvis @SimonKrughoff @pmelchior I've implemented the mask subselection and color-coded by mask bit. I've kept the mask plane opaque, but ordered the mask bits (BAD > CR > EDGE > INTRP > SAT > NO_DATA).
1. Does this subset of bits make sense?
2. Do we want to keep the color coding?

Example images can be found on the development version of the exposure checker:
https://portal-auth.nersc.gov/projecta/lsst/dev-exp-checker/viewer.html?release=r1.1&expname=v417009-fr&ccd=22-10-A

**Comment by rmjarvis** [comment]
I tried the developer version, and at least for me, Toggle Mask doesn't seem to be working.  If I hit Toggle Mask, nothing changes.  And then if I try to do Skip, it goes into spinning circle mode, not loading the next image.

**Comment by kadrlica** [comment]
Ah, sorry, code cleanup resulted in a misplaced  comma. Should be fixed now.

**Comment by rmjarvis** [comment]
I think having colors is nice.  It seems useful to know that a masked pixel is for a cosmic ray rather than a hot pixel or something.  However, I'd like to rearrange some of the color assignments slightly.  

* 0: BAD (red)  I didn't see any of these.  Are there any?
* 1: SAT (green)  This one is good.
* 3: CR (magenta)  I found the magenta a bit hard to notice.  Probably since these are usually quite small, but I think maybe yellow or blue would be better here.
* 4: EDGE (yellow)  This could be magenta or even grey.  Something less obvious, since it's always in the same place, so we don't need it to catch the eye so much.  The yellow here is a bit distracting I think.
* 8: NO_DATA (orange)  I didn't see any of these either.

**Comment by kadrlica** [comment]
@rmjarvis thanks for taking a look. I'll let @rhl comment on color, but it was previously suggested that we stick with the same color scheme as DM.

**Comment by rmjarvis** [comment]
> it was previously suggested that we stick with the same color scheme as DM.

FWIW, I can't see any advantage to this.  We should use whatever colors we think will produce the best results for our purpose.  The number of users who will have any familiarity with the DM color scheme will be pretty close to zero.

**Comment by SimonKrughoff** [comment]
Sorry for not being more involved with this issue.  I'm pretty much in agreement with what's been discussed here.  My comments are:
1) I agree that most people will not be used to the DM color scheme, but it won't be zero.  I'm not arguing against using a different scheme if it helps, but if there is no compelling reason to use a different one would suggest sticking with DM's.
2) I have, in the past, found the `INTRP` bit useful.  It helps with trying to diagnose things like what Dominique was seeing in the bleed trails.  It's not essential, if it makes things too busy.

**Comment by kadrlica** [comment]
Thanks @SimonKrughoff. Is it true that pixels with `INTRP` set will also have another bit set?

**Comment by SimonKrughoff** [comment]
If they don't, there is a problem.

**Comment by kadrlica** [comment]
Ok, I can put `INTRP` back in. It won't show up unless something goes wrong (the masks are opaque).

**Comment by kadrlica** [comment]
Pixel subsets and coloring has been implemented.
